src/lib/auth-utils.ts

```diff
@@ -16,7 +22,7 @@ export function validatePassword(password: string, minLength: number = DEFAULT_M
  * Create a RemoteUser object
  * @return {Object} { name: xx, pluginGroups: [], real_groups: [] }
  */
-export function createRemoteUser(name: string, pluginGroups: string[]): RemoteUser {
+export function createRemoteUser(name: string | void, pluginGroups: string[]): RemoteUser {
```

Comment:
This not correct, the `createRemoteUser` must have a name, in the case is not available for such reason we have `createAnonymousRemoteUser()`

Reply:
this is just a fix for a type error in the existing verdaccio logic, if the name is mandatory it needs to specified as such in the RemoteUser interface used by the credentials object in https://github.com/verdaccio/verdaccio/blob/727d683d197dc36dcf9672a3a8a501559851b8b6/src/lib/auth.ts#L434 see https://github.com/verdaccio/monorepo/blob/master/core/types/index.d.ts#L26 or should be pre-validated before it is passed to the createRemoteUser function. I assume the latter would be a preferable fix